tests/beta/test_default_tensor_tf.py

```diff
@@ -679,10 +689,10 @@ def test_torch_interface(self, cost, interface, diff_method, tol):
         assert np.allclose(res.detach().numpy(), self.expected_grad, atol=tol, rtol=0)
 
     @pytest.mark.parametrize("interface", ["tf"])
-    @pytest.mark.parametrize("diff_method", ["device"])
-    def test_tf_interface_device_diff(self, cost, interface, diff_method, tol):
-        """Tests that the gradient of an arbitrary U3 gate is correct
-        using the TensorFlow interface"""
+    @pytest.mark.parametrize("diff_method", ["backprop", "device"])
+    def test_tf_interface(self, cost, interface, diff_method, tol):
+        """Tests that the gradient of an arbitrary U3 gate is correct using the
+        TensorFlow interface with the allowed differentiation methods"""
```

Comment:
With #631 merged, we could create a single test case for both diff methods (`"backprop"` is only supported for the `tf` interface). Therefore the code related to using a cost with decomposition and the related test case could be removed.